### PR TITLE
gapis: Filter SubcommandGroups too

### PR DESCRIPTION
**Pull request by ben-clayton**
These were being added with no consideration for the filtering.

To implement this I changed `CmdIDGroup.AddAtoms()` into `AddCommand()` and `Cluster()`.
This is a cleaner API anyway.

### CODE REVIEW

**Comment by ben-clayton** [comment]
(I just threw in another related CL)

**Comment by Qining** [comment]
LGTM

File: gapis/api/cmd_id_group.go [hunk 1]
```diff
@@ -443,54 +442,68 @@ func (g CmdIDGroup) FindSubCommandRoot(id CmdID) *SubCmdRoot {
...
-
-	scan := func(to CmdID) {
-		for id := rng.Start; id < to; id++ {
-			if !pred(id) {
-				rng.End = id
-				if rng.Start != rng.End {
-					spans = append(spans, &CmdIDRange{rng.Start, rng.End})
-				}
-				rng.Start = id + 1
+// AddCommand adds the command to the groups.
```
Comment (Qining) [inline gapis/api/cmd_id_group.go]: groups -> group

Comment (ben-clayton) [inline gapis/api/cmd_id_group.go]: I meant groups. It can propagate down to sub-groups. I admit I could expand on this a little more.

Comment (Qining) [inline gapis/api/cmd_id_group.go]: Got it. Thanks.

**Pull request merged**